### PR TITLE
Add OAuth 2.0 authorization code support for Apigee demos

### PR DESCRIPTION
**Pull request by drush1980**
### Background 
Adds the ability to support OAuth authorization code grant type requests from an authorization server such as Apigee.

### Change Summary
- Updated frontend to be able to process an OAuth based login request
- Added a consent form
- Added a ConfigMap to the extras directory to store the OAuth client ID and redirect URL

### CODE REVIEW

File: extras/apigee/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+[Apigee Monetization](https://cloud.google.com/apigee/docs/api-platform/monetization/overview) allows you to build paid rate plans to monetize access to API products.
+
+Bank of Anthos supports the ability to authorize external access requests made using an [OAuth 2.0 Authorization Code grant type](https://oauth.net/2/grant-types/authorization-code/). Apigee can act as the [authorization server](https://cloud.google.com/apigee/docs/api-platform/security/oauth/oauth-home) that issues and validates the access tokens used to call the APIs securely. This directory contains a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) that defines an allowed OAuth client ID and redirect URI. The frontend service will use these to validate incoming OAuth requests. These values should be replaced with a client ID and redirect URI that is defined in an Apigee [API proxy](https://cloud.google.com/apigee/docs/api-platform/fundamentals/understanding-apis-and-api-proxies) that implements the [authorization code grant type](https://cloud.google.com/apigee/docs/api-platform/security/oauth/oauth-v2-policy-authorization-code-grant-type).
+
+## Prerequisites
```
Comment (bourgeoisor) [inline extras/apigee/README.md:13]: A couple steps after the prereqs that shows how you deploy BoA with Apigee would help a lot

File: extras/apigee/oauth-config.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# limitations under the License.
+
+# [START gke_boa_kubernetes_manifests_config_configmap_oauth_config]
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: oauth-config
+data:
+  DEMO_OAUTH_CLIENT_ID: "{APIGEE_CLIENT_ID}"
+  DEMO_OAUTH_REDIRECT_URI: "https://{APIGEE_DOMAIN}/oauth/callback"
```
Comment (bourgeoisor) [inline extras/apigee/oauth-config.yaml:22]: Are these substituted in manually? If so, this step should be in the README at minimum

File: kubernetes-manifests/frontend.yaml [hunk 3]
```diff
@@ -66,6 +66,18 @@ spec:
             configMapKeyRef:
               name: demo-data-config
               key: DEMO_LOGIN_PASSWORD
+        - name: REGISTERED_OAUTH_CLIENT_ID
```
Comment (bourgeoisor) [inline kubernetes-manifests/frontend.yaml:69]: You can remove the diff from `kubernetes-manifest`, it's going to happen once I release a new version of Bank of Anthos (this directory is essentially "latest release")

**Review by bourgeoisor — CHANGES_REQUESTED** [review]
Checks are failing, make sure to lint the code changes

File: extras/apigee/oauth-config.yaml (hunk 2 above)
Comment (drush1980) [reply extras/apigee/oauth-config.yaml:22]: This is explained in line 11 of the README

File: extras/apigee/README.md (hunk 1 above)
Comment (drush1980) [reply extras/apigee/README.md:13]: Planning to add this once the full set of accompanying Apigee assets are ready to be committed here. Intent was to address that in a future PR.

File: kubernetes-manifests/frontend.yaml (hunk 3 above)
Comment (drush1980) [reply kubernetes-manifests/frontend.yaml]: Done

**Review by bourgeoisor — APPROVED** [review]
LGTM

**Pull request merged**